plugins/file_selector/file_selector_linux/linux/test/file_selector_plugin_test.cc

```diff
@@ -0,0 +1,166 @@
+  g_autoptr(GSList) type_group_list =
+      gtk_file_chooser_list_filters(GTK_FILE_CHOOSER(dialog));
+  EXPECT_EQ(g_slist_length(type_group_list), 3);
+  GtkFileFilter* text_filter =
+      GTK_FILE_FILTER(g_slist_nth_data(type_group_list, 0));
+  GtkFileFilter* image_filter =
+      GTK_FILE_FILTER(g_slist_nth_data(type_group_list, 1));
+  GtkFileFilter* any_filter =
+      GTK_FILE_FILTER(g_slist_nth_data(type_group_list, 2));
+  // Filters can't be inspected, so query them to see that they match expected
+  // filter behavior.
+  GtkFileFilterInfo text_file_info = {};
+  text_file_info.contains = static_cast<GtkFileFilterFlags>(
+      GTK_FILE_FILTER_DISPLAY_NAME | GTK_FILE_FILTER_MIME_TYPE);
+  text_file_info.display_name = "foo.txt";
+  text_file_info.mime_type = "text/plain";
+  GtkFileFilterInfo image_file_info = {};
+  image_file_info.contains = static_cast<GtkFileFilterFlags>(
+      GTK_FILE_FILTER_DISPLAY_NAME | GTK_FILE_FILTER_MIME_TYPE);
+  image_file_info.display_name = "foo.png";
+  image_file_info.mime_type = "image/png";
+  EXPECT_TRUE(gtk_file_filter_filter(text_filter, &text_file_info));
+  EXPECT_FALSE(gtk_file_filter_filter(text_filter, &image_file_info));
+  EXPECT_FALSE(gtk_file_filter_filter(image_filter, &text_file_info));
+  EXPECT_TRUE(gtk_file_filter_filter(image_filter, &image_file_info));
+  EXPECT_TRUE(gtk_file_filter_filter(any_filter, &image_file_info));
+  EXPECT_TRUE(gtk_file_filter_filter(any_filter, &text_file_info));
+}
+
+TEST(FileSelectorPlugin, TestSaveSimple) {
+  g_autoptr(FlValue) args = fl_value_new_map();
+
+  g_autoptr(GtkFileChooserNative) dialog =
+      create_dialog_for_method(nullptr, "getSavePath", args);
+
+  ASSERT_NE(dialog, nullptr);
+  EXPECT_EQ(gtk_file_chooser_get_action(GTK_FILE_CHOOSER(dialog)),
+            GTK_FILE_CHOOSER_ACTION_SAVE);
+  EXPECT_EQ(gtk_file_chooser_get_select_multiple(GTK_FILE_CHOOSER(dialog)),
+            false);
+}
+
+TEST(FileSelectorPlugin, TestSaveWithArguments) {
+  g_autoptr(FlValue) args = fl_value_new_map();
+  fl_value_set_string_take(args, "initialDirectory",
+                           fl_value_new_string("/tmp"));
+  fl_value_set_string_take(args, "suggestedName",
+                           fl_value_new_string("foo.txt"));
+
+  g_autoptr(GtkFileChooserNative) dialog =
+      create_dialog_for_method(nullptr, "getSavePath", args);
+
+  ASSERT_NE(dialog, nullptr);
+  EXPECT_EQ(gtk_file_chooser_get_action(GTK_FILE_CHOOSER(dialog)),
+            GTK_FILE_CHOOSER_ACTION_SAVE);
+  EXPECT_EQ(gtk_file_chooser_get_select_multiple(GTK_FILE_CHOOSER(dialog)),
+            false);
+  g_autofree gchar* current_name =
+      gtk_file_chooser_get_current_name(GTK_FILE_CHOOSER(dialog));
+  EXPECT_STREQ(current_name, "foo.txt");
+  // TODO(stuartmorgan): gtk_file_chooser_get_current_folder doesn't seem to
+  // return a value set by gtk_file_chooser_set_current_folder, or at least
+  // doesn't in a test context, so that's not currently validated.
```

Comment:
What does it return? Looks like it's meant to return null if it was unable to load the directory passed to `gtk_file_chooser_set_current_folder`.

Reply:
I believe it was null. My guess is that it's because the dialog hasn't actually been shown yet, so

> Note that this is the folder that the file chooser is currently displaying

is arguably saying that null is correct behavior before showing it.